modules/dnn/test/test_onnx_conformance.cpp

```diff
@@ -0,0 +1,1250 @@
+    {"test_transpose_default", 1, 1},
+    {"test_tril", 1, 1},
+    {"test_tril_neg", 2, 1},
+    {"test_tril_one_row_neg", 1, 1},
+    {"test_tril_out_neg", 2, 1},
+    {"test_tril_out_pos", 2, 1},
+    {"test_tril_pos", 2, 1},
+    {"test_tril_square", 1, 1},
+    {"test_tril_square_neg", 2, 1},
+    {"test_tril_zero", 2, 1},
+    {"test_triu", 1, 1},
+    {"test_triu_neg", 2, 1},
+    {"test_triu_one_row", 2, 1},
+    {"test_triu_out_neg_out", 2, 1},
+    {"test_triu_out_pos", 2, 1},
+    {"test_triu_pos", 2, 1},
+    {"test_triu_square", 1, 1},
+    {"test_triu_square_neg", 2, 1},
+    {"test_triu_zero", 2, 1},
+    {"test_unique_not_sorted_without_axis", 1, 4},
+    {"test_unique_sorted_with_axis", 1, 4},
+    {"test_unique_sorted_with_axis_3d", 1, 4},
+    {"test_unique_sorted_with_negative_axis", 1, 4},
+    {"test_unique_sorted_without_axis", 1, 4},
+    {"test_unsqueeze_axis_0", 2, 1},
+    {"test_unsqueeze_axis_1", 2, 1},
+    {"test_unsqueeze_axis_2", 2, 1},
+    {"test_unsqueeze_axis_3", 1, 1},
+    {"test_unsqueeze_negative_axes", 2, 1},
+    {"test_unsqueeze_three_axes", 2, 1},
+    {"test_unsqueeze_two_axes", 2, 1},
+    {"test_unsqueeze_unsorted_axes", 2, 1},
+    {"test_upsample_nearest", 2, 1},
+    {"test_where_example", 3, 1},
+    {"test_where_long_example", 3, 1},
+    {"test_xor2d", 2, 1},
+    {"test_xor3d", 2, 1},
+    {"test_xor4d", 2, 1},
+    {"test_xor_bcast3v1d", 2, 1},
+    {"test_xor_bcast3v2d", 2, 1},
+    {"test_xor_bcast4v2d", 2, 1},
+    {"test_xor_bcast4v3d", 2, 1},
+    {"test_xor_bcast4v4d", 2, 1},
+};
+
+
+struct TestCaseInput
+{
+    std::vector<std::string> input_paths;
+    std::vector<std::string> output_paths;
+    std::string model_path;
+    std::string name;
+};
+
+std::ostream& operator<<(std::ostream& os, const TestCaseInput& test_case)
+{
+    return os << test_case.name;
+}
+
+typedef tuple<TestCaseInput, tuple<Backend, Target> > ONNXConfParams;
+
+std::string printOnnxConfParams(const testing::TestParamInfo<ONNXConfParams>& params)
+{
+    TestCaseInput test_case = get<0>(params.param);
+    Backend backend = get<0>(get<1>(params.param));
+    Target target = get<1>(get<1>(params.param));
+
+    std::stringstream ss;
+    ss << test_case.name << "_";
+    PrintTo(backend, &ss);
+    ss << "_";
+    PrintTo(target, &ss);
+
+    return ss.str();
+}
+
+template<typename TString>
+static std::string _tf(TString filename, bool required = true)
+{
+    return findDataFile(std::string("dnn/onnx/") + filename, required);
+}
+
+std::vector<TestCaseInput> readTestCases()
+{
+    std::vector<TestCaseInput> ret;
+    for (size_t i = 0; i < sizeof(testConformanceConfig) / sizeof(testConformanceConfig[0]); ++i)
+    {
+        const TestCase& test_case = testConformanceConfig[i];
+
+        TestCaseInput input;
+
+        std::string prefix = cv::format("conformance/node/%s", test_case.name);
+        input.name = test_case.name;
+        input.model_path = _tf(cv::format("%s/model.onnx", prefix.c_str()));
```

Comment:
This approach terminates whole test app if data is missing.

```
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(3.4.16-dev) /home/alalek/projects/opencv/dev/modules/ts/src/ts.cpp:1062: error: (-2:Unspecified error) OpenCV tests: Can't find required data file: dnn/onnx/conformance/node/test_abs/model.onnx in function 'findData'
```

As result zero tests are launched including tests without external test data.

---

I will fix that with filter updates.